oauthproxy.go

```diff
@@ -589,6 +599,27 @@ func (p *OAuthProxy) GetRedirect(req *http.Request) (redirect string, err error)
 	return
 }
 
+// getRedirectFromForwardHeaders returns the redirect URL based on X-Forwarded-{Proto,Host,Port,Uri} headers
+func (p *OAuthProxy) getRedirectFromForwardHeaders(req *http.Request) string {
+	proto := fmt.Sprintf("%s://", util.GetRequestProto(req))
+
+	host, port := splitHostPort(util.GetRequestHost(req))
+	port = util.GetRequestPortFromHeader(req, port)
+	formedPort := fmt.Sprintf(":%s", port)
+
+	uri := util.GetRequestURI(req)
+
+	if port != "" && validOptionalPort(formedPort) {
+		port = formedPort
+	}
+
+	if strings.HasPrefix(uri, p.ProxyPrefix) {
```

Comment:
I think we need to handle the trailing slash (that is also missing from the above usage). For instance if my prefix is set to `/oauth2` - a custom custom route structure in my app as `/oauth2foobar` should not match that.

```suggestion
	if strings.HasPrefix(uri, fmt.Springf("%s/", p.ProxyPrefix)) {
```